lib/concurrent/delay.rb

```diff
@@ -145,16 +147,15 @@ def wait(timeout = nil)
     # @yield the delayed operation to perform
     # @return [true, false] if success
     def reconfigure(&block)
-      mutex.lock
-      raise ArgumentError.new('no block given') unless block_given?
-      unless @computing
-        @task = block
-        true
-      else
-        false
+      mutex.synchronize do
+        raise ArgumentError.new('no block given') unless block_given?
+        unless @computing
+          @task = block
+          true
+        else
+          false
+        end
```

Comment:
Since delay is migrated to `Delay < Synchronization::Object` it probably should be using its method `synchronize`, is there a reason for using `mutex` which I am missing?